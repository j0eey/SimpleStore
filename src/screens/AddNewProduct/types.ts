import { Asset } from "react-native-image-picker";
import { LocationType } from "../../types/types";

export interface FormHeaderProps {
  title: string;
}

export interface ImageUploadSectionProps {
  images: Asset[];
  onImagePick: () => void;
  onCameraCapture: () => void;
  onRemoveImage: (index: number) => void;
  isSubmitting: boolean;
}

export interface FormInputProps {
  label: string;
  value: string;
  onChangeText: (text: string) => void;
  placeholder: string;
  multiline?: boolean;
  numberOfLines?: number;
  keyboardType?: 'default' | 'numeric';
  isSubmitting: boolean;
}

export interface LocationSelectorProps {
  location: LocationType;
  onLocationSelect: () => void;
  isSubmitting: boolean;
}

export interface FormActionsProps {
  onSubmit: () => void;
  onCancel: () => void;
  isSubmitting: boolean;
}