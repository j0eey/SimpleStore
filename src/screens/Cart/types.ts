import { Animated } from "react-native";
import { CartItemType } from "../../types/Cart";

export interface CartHeaderProps {
  itemCount: number;
}

export interface EmptyCartProps {
  onStartShopping: () => void;
}

export interface CartItemProps {
  item: CartItemType;
  onIncreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onDecreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onRemoveItem: (itemId: string) => void;
}

export interface CartSummaryProps {
  cart: CartItemType[];
  summaryExpanded: boolean;
  summaryHeight: Animated.Value;
  onToggleSummary: () => void;
  onCheckout: () => void;
  onClearCart: () => void;
  calculateTotal: () => string;
}

export interface CartListProps {
  cart: CartItemType[];
  onIncreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onDecreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onRemoveItem: (itemId: string) => void;
}