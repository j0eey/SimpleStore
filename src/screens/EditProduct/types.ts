import { Asset } from "react-native-image-picker";
import { LocationType } from "../../types/types";


export interface LoadingScreenProps {}

export interface FormHeaderProps {
  title: string;
}

export interface ImageUploadSectionProps {
  canAddMoreImages: boolean;
  isSubmitting: boolean;
  onGalleryPress: () => void;
  onCameraPress: () => void;
}

export interface ImagePreviewSectionProps {
  existingImages: string[];
  imagesToDelete: string[];
  newImages: Asset[];
  isSubmitting: boolean;
  onRemoveExisting: (imageUrl: string) => void;
  onRemoveNew: (index: number) => void;
}

export interface FormFieldProps {
  label: string;
  value: string;
  onChangeText: (text: string) => void;
  placeholder: string;
  multiline?: boolean;
  numberOfLines?: number;
  keyboardType?: 'default' | 'numeric';
  editable?: boolean;
}

export interface LocationPickerProps {
  location: LocationType;
  isSubmitting: boolean;
  onPress: () => void;
}

export interface FormActionsProps {
  isSubmitting: boolean;
  onSubmit: () => void;
  onCancel: () => void;
  disabled: boolean;
}
