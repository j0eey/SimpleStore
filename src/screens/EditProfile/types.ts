import { EditableProfile } from "../../types/types";

export interface ProfileImageProps {
  imageUrl?: string;
  onPress?: () => void;
}

export interface EditFormProps {
  editedProfile: EditableProfile;
  onProfileChange: (profile: EditableProfile) => void;
  onSave: () => void;
  onCancel: () => void;
  updating: boolean;
}

export interface ImagePickerModalProps {
  visible: boolean;
  onClose: () => void;
  onSelectFromCamera: () => void;
  onSelectFromLibrary: () => void;
}

export interface ProfileSectionProps {
  children: React.ReactNode;
}

export interface MainContainerProps {
  children: React.ReactNode;
}