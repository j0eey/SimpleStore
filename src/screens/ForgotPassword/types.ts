import { Control, FieldErrors } from "react-hook-form";

export interface FormData {
  email: string;
}

export interface FormHeaderProps {
  title: string;
}

export interface EmailInputProps {
  control: Control<FormData>;
  errors: FieldErrors<FormData>;
}

export interface SubmitButtonProps {
  onPress: () => void;
  loading: boolean;
}

export interface LoginLinkProps {
  onPress: () => void;
}

export interface MainContainerProps {
  children: React.ReactNode;
}