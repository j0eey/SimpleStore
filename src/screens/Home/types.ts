import { Product } from "../../types/Product";

export interface ProductItemProps {
  item: Product;
  onPress: (product: Product) => void;
  onAddToCart: (product: Product, position: {x: number, y: number, width: number, height: number}) => void;
  isOwner: boolean;
}

export interface CategoryListProps {
  loading: boolean;
  onCategoryPress: (categoryName: string) => void;
}

export interface LocationSelectorProps {
  selectedLocation: string;
  onSelect: (location: string) => void;
}

export interface ErrorDisplayProps {
  errorMessage: string;
  onRetry: () => void;
}

export interface HomeHeaderProps {
  onProfilePress: () => void;
}