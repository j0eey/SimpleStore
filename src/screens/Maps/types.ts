import { MapPressEvent, Region } from "react-native-maps";
import { Prediction } from "../../types/types";

export interface LocationData {
  name: string;
  latitude: number;
  longitude: number;
}

export interface HeaderProps {
  onGoBack: () => void;
}

export interface SearchBarProps {
  searchText: string;
  onSearchChange: (text: string) => void;
  onSearchClear: () => void;
  onSearchFocus: () => void;
}

export interface PredictionsListProps {
  predictions: Prediction[];
  showPredictions: boolean;
  onPredictionSelect: (placeId: string, description: string) => void;
}

export interface MapContainerProps {
  region: Region;
  selectedLocation: LocationData | null;
  onRegionChange: (region: Region) => void;
  onMapPress: (event: MapPressEvent) => void;
}

export interface ApplyButtonProps {
  selectedLocation: LocationData | null;
  onApply: () => void;
}

export interface MainContainerProps {
  children: React.ReactNode;
}