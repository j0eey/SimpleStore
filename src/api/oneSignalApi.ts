import Config from 'react-native-config';
import { OneSignalNotificationPayload, OneSignalApiResponse } from "../types/Product";


class OneSignalApi {
  private restApiKey = Config.ONESIGNAL_REST_API_KEY;
  private baseUrl = 'https://api.onesignal.com';

  /**
   * Send notification via OneSignal REST API
   */
  async sendNotification(payload: OneSignalNotificationPayload): Promise<OneSignalApiResponse> {
    const response = await fetch(`${this.baseUrl}/notifications`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Key ${this.restApiKey}`,
      },
      body: JSON.stringify(payload),
    });

    if (!response.ok) {
      throw new Error(`OneSignal API error: ${response.status} ${response.statusText}`);
    }

    return await response.json();
  }
}

export default new OneSignalApi();