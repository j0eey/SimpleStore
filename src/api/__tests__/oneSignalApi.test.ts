import oneSignalApi from '../oneSignalApi';
import { OneSignalNotificationPayload, OneSignalApiResponse } from '../../types/Product';

// Mock react-native-config
jest.mock('react-native-config', () => ({
  ONESIGNAL_REST_API_KEY: 'test-rest-api-key-12345',
}));

// Mock global fetch
const mockFetch = jest.fn();
(globalThis as any).fetch = mockFetch;

describe('OneSignalApi', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const mockNotificationPayload: OneSignalNotificationPayload = {
    app_id: 'test-app-id',
    included_segments: ['All'],
    headings: { en: 'Test Notification' },
    contents: { en: 'This is a test notification' },
    data: {
      productId: 'product-123',
      productTitle: 'Test Product',
      type: 'product_added',
      price: '99.99',
      location: 'New York, NY',
    },
    buttons: [
      { id: 'view_product', text: 'View Product' },
      { id: 'dismiss', text: 'Dismiss' },
    ],
    big_picture: 'https://example.com/image.jpg',
    ios_attachments: {
      product_image: 'https://example.com/image.jpg',
    },
  };

  const mockSuccessResponse: OneSignalApiResponse = {
    id: 'notification-id-123',
    recipients: 150,
    external_id: null,
  };

  describe('sendNotification', () => {
    describe('Successful Requests', () => {
      it('should send notification successfully', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith('https://api.onesignal.com/notifications', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Authorization': 'Key test-rest-api-key-12345',
          },
          body: JSON.stringify(mockNotificationPayload),
        });

        expect(result).toEqual(mockSuccessResponse);
      });

      it('should use correct OneSignal API endpoint', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          'https://api.onesignal.com/notifications',
          expect.any(Object)
        );
      });

      it('should include correct headers', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          expect.any(String),
          expect.objectContaining({
            headers: {
              'Content-Type': 'application/json',
              'Authorization': 'Key test-rest-api-key-12345',
            },
          })
        );
      });

      it('should send payload as JSON string in body', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          expect.any(String),
          expect.objectContaining({
            method: 'POST',
            body: JSON.stringify(mockNotificationPayload),
          })
        );
      });

      it('should handle different response data structures', async () => {
        const differentResponse = {
          id: 'different-id',
          recipients: 250,
          external_id: 'external-123',
          errors: null,
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(differentResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(result).toEqual(differentResponse);
      });
    });

    describe('Error Handling', () => {
      it('should throw error for 400 Bad Request', async () => {
        const mockResponse = {
          ok: false,
          status: 400,
          statusText: 'Bad Request',
          json: jest.fn(),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('OneSignal API error: 400 Bad Request');

        expect(mockResponse.json).not.toHaveBeenCalled();
      });

      it('should throw error for 401 Unauthorized', async () => {
        const mockResponse = {
          ok: false,
          status: 401,
          statusText: 'Unauthorized',
          json: jest.fn(),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('OneSignal API error: 401 Unauthorized');
      });

      it('should throw error for 429 Rate Limited', async () => {
        const mockResponse = {
          ok: false,
          status: 429,
          statusText: 'Too Many Requests',
          json: jest.fn(),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('OneSignal API error: 429 Too Many Requests');
      });

      it('should throw error for 500 Internal Server Error', async () => {
        const mockResponse = {
          ok: false,
          status: 500,
          statusText: 'Internal Server Error',
          json: jest.fn(),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('OneSignal API error: 500 Internal Server Error');
      });

      it('should handle network errors', async () => {
        const networkError = new Error('Network request failed');
        mockFetch.mockRejectedValue(networkError);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('Network request failed');
      });

      it('should handle JSON parsing errors', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockRejectedValue(new Error('Invalid JSON')),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('Invalid JSON');
      });

      it('should handle timeout errors', async () => {
        const timeoutError = new Error('Request timeout');
        mockFetch.mockRejectedValue(timeoutError);

        await expect(oneSignalApi.sendNotification(mockNotificationPayload))
          .rejects.toThrow('Request timeout');
      });
    });

    describe('Payload Variations', () => {
      it('should handle minimal notification payload', async () => {
        const minimalPayload: OneSignalNotificationPayload = {
          app_id: 'test-app-id',
          included_segments: ['All'],
          headings: { en: 'Simple Notification' },
          contents: { en: 'Simple content' },
          data: {
            type: 'minimal_notification',
          },
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(minimalPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          'https://api.onesignal.com/notifications',
          expect.objectContaining({
            body: JSON.stringify(minimalPayload),
          })
        );
        expect(result).toEqual(mockSuccessResponse);
      });

      it('should handle notification with multiple languages', async () => {
        const multiLanguagePayload: OneSignalNotificationPayload = {
          app_id: 'test-app-id',
          included_segments: ['All'],
          headings: { en: 'English Title' },
          contents: { en: 'English content' },
          data: {
            type: 'multilingual_test',
          },
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(multiLanguagePayload);

        expect(mockFetch).toHaveBeenCalledWith(
          'https://api.onesignal.com/notifications',
          expect.objectContaining({
            body: JSON.stringify(multiLanguagePayload),
          })
        );
      });

      it('should handle notification without buttons', async () => {
        const noButtonsPayload = {
          ...mockNotificationPayload,
          buttons: undefined,
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(noButtonsPayload);

        const sentPayload = JSON.parse(mockFetch.mock.calls[0][1].body);
        expect(sentPayload.buttons).toBeUndefined();
      });

      it('should handle notification without images', async () => {
        const noImagesPayload = {
          ...mockNotificationPayload,
          big_picture: undefined,
          ios_attachments: undefined,
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(noImagesPayload);

        const sentPayload = JSON.parse(mockFetch.mock.calls[0][1].body);
        expect(sentPayload.big_picture).toBeUndefined();
        expect(sentPayload.ios_attachments).toBeUndefined();
      });

      it('should handle notification with custom data', async () => {
        const customDataPayload = {
          ...mockNotificationPayload,
          data: {
            customField1: 'value1',
            customField2: 'value2',
            userId: '12345',
            actionType: 'custom_action',
          },
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(customDataPayload);

        const sentPayload = JSON.parse(mockFetch.mock.calls[0][1].body);
        expect(sentPayload.data).toEqual(customDataPayload.data);
      });
    });

    describe('Configuration and Environment', () => {
      it('should use REST API key from config', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          expect.any(String),
          expect.objectContaining({
            headers: expect.objectContaining({
              'Authorization': 'Key test-rest-api-key-12345',
            }),
          })
        );
      });

      it('should use correct OneSignal base URL', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          'https://api.onesignal.com/notifications',
          expect.any(Object)
        );
      });

      it('should use POST method', async () => {
        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(mockFetch).toHaveBeenCalledWith(
          expect.any(String),
          expect.objectContaining({
            method: 'POST',
          })
        );
      });
    });

    describe('Response Handling', () => {
      it('should return response data as-is', async () => {
        const customResponse = {
          id: 'custom-notification-id',
          recipients: 999,
          external_id: 'external-custom',
          errors: null,
          warnings: [],
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(customResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(result).toEqual(customResponse);
      });

      it('should handle empty response', async () => {
        const emptyResponse = {};

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(emptyResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(result).toEqual(emptyResponse);
      });

      it('should handle response with errors', async () => {
        const errorResponse = {
          id: '',
          recipients: 0,
          errors: ['Invalid API key', 'Missing required field'],
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(errorResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(mockNotificationPayload);

        expect(result).toEqual(errorResponse);
      });
    });

    describe('Edge Cases', () => {
      it('should handle very large payload', async () => {
        const largePayload = {
          ...mockNotificationPayload,
          contents: { 
            en: 'A'.repeat(1000), // Very long content
          },
          data: {
            ...mockNotificationPayload.data,
            largeField: 'B'.repeat(2000),
          },
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        const result = await oneSignalApi.sendNotification(largePayload);

        expect(result).toEqual(mockSuccessResponse);
      });

      it('should handle special characters in payload', async () => {
        const specialCharsPayload = {
          ...mockNotificationPayload,
          headings: { en: 'Test 🎉 Special chars: & < > " \' \\' },
          contents: { en: 'Content with émojis 😀 and spëcial chârs' },
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(specialCharsPayload);

        const sentPayload = JSON.parse(mockFetch.mock.calls[0][1].body);
        expect(sentPayload.headings.en).toContain('🎉');
        expect(sentPayload.contents.en).toContain('😀');
      });

      it('should handle null and undefined values in payload', async () => {
        const payloadWithNulls = {
          ...mockNotificationPayload,
          big_picture: undefined,
          ios_attachments: undefined,
        };

        const mockResponse = {
          ok: true,
          status: 200,
          json: jest.fn().mockResolvedValue(mockSuccessResponse),
        };
        mockFetch.mockResolvedValue(mockResponse);

        await oneSignalApi.sendNotification(payloadWithNulls);

        expect(mockFetch).toHaveBeenCalledWith(
          'https://api.onesignal.com/notifications',
          expect.objectContaining({
            body: JSON.stringify(payloadWithNulls),
          })
        );
      });
    });
  });
});