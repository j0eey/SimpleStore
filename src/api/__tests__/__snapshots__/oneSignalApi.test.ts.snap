// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneSignalApi Request Payload Snapshots should match snapshot for minimal notification payload: minimal-notification-payload 1`] = `
{
  "app_id": "test-app-id",
  "contents": {
    "en": "Simple content",
  },
  "data": {
    "type": "minimal_notification",
  },
  "headings": {
    "en": "Simple Notification",
  },
  "included_segments": [
    "All",
  ],
}
`;

exports[`OneSignalApi Request Payload Snapshots should match snapshot for notification with custom data: custom-data-notification-payload 1`] = `
{
  "app_id": "test-app-id",
  "big_picture": "https://example.com/image.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "This is a test notification",
  },
  "data": {
    "actionType": "custom_action",
    "customField1": "value1",
    "customField2": "value2",
    "userId": "12345",
  },
  "headings": {
    "en": "Test Notification",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/image.jpg",
  },
}
`;

exports[`OneSignalApi Request Payload Snapshots should match snapshot for notification with special characters: special-characters-payload 1`] = `
{
  "app_id": "test-app-id",
  "big_picture": "https://example.com/image.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "Content with Ã©mojis ðŸ˜€ and spÃ«cial chÃ¢rs",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Test Product",
    "type": "product_added",
  },
  "headings": {
    "en": "Test ðŸŽ‰ Special chars: & < > " ' \\",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/image.jpg",
  },
}
`;

exports[`OneSignalApi Request Payload Snapshots should match snapshot for notification without optional fields: no-optional-fields-payload 1`] = `
{
  "app_id": "test-app-id",
  "contents": {
    "en": "This is a test notification",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Test Product",
    "type": "product_added",
  },
  "headings": {
    "en": "Test Notification",
  },
  "included_segments": [
    "All",
  ],
}
`;

exports[`OneSignalApi Request Payload Snapshots should match snapshot for request headers and config: request-configuration 1`] = `
{
  "headers": {
    "Authorization": "Key test-rest-api-key-12345",
    "Content-Type": "application/json",
  },
  "method": "POST",
  "url": "https://api.onesignal.com/notifications",
}
`;

exports[`OneSignalApi Request Payload Snapshots should match snapshot for standard notification payload: standard-notification-payload 1`] = `
{
  "app_id": "test-app-id",
  "big_picture": "https://example.com/image.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "This is a test notification",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Test Product",
    "type": "product_added",
  },
  "headings": {
    "en": "Test Notification",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/image.jpg",
  },
}
`;

exports[`OneSignalApi Response Structure Snapshots should match snapshot for response with additional fields: extended-response 1`] = `
{
  "errors": null,
  "external_id": "external-123",
  "id": "notification-id-123",
  "recipients": 150,
  "warnings": [],
}
`;

exports[`OneSignalApi Response Structure Snapshots should match snapshot for response with errors: error-response 1`] = `
{
  "errors": [
    "Invalid API key",
    "Missing required field",
  ],
  "id": "",
  "recipients": 0,
}
`;

exports[`OneSignalApi Response Structure Snapshots should match snapshot for successful response: success-response 1`] = `
{
  "external_id": null,
  "id": "notification-id-123",
  "recipients": 150,
}
`;
